src/router.js

```diff
@@ -1,72 +1,64 @@
-export function router(options) {
-  return function(emit) {
-    return {
-      state: {
-        router: {}
-      },
-      actions: {
-        router: {
-          set: function(state, actions, data) {
-            return {
-              router: data
-            }
-          },
-          go: function(state, actions, path) {
-            if (location.pathname + location.search !== path) {
-              history.pushState({}, "", path)
-              actions.router.set({
-                path: path
-              })
-            }
-          }
-        }
-      },
-      events: {
-        load: function(state, actions) {
-          addEventListener("popstate", function() {
-            actions.router.set({})
-          })
-        },
-        render: function(state, actions, view) {
-          return view[
-            (state.router.index >= 0
-              ? state
-              : actions.router.set(
-                  emit("route", match(location.pathname, view))
-                )).router.index
-          ][1]
-        }
+import { h } from 'hyperapp'
+
+// router module
+export const router = {
+  actions: {
+    set: function (state, actions, data) {
+      return data
+    },
+    go: function (state, actions, path) {
+      if (location.pathname + location.search !== path) {
+        history.pushState({}, '', path)
+        actions.set({ path: path })
       }
     }
+  },
+  init(state, actions) {
+    window.addEventListener('popstate', function () {
```

Comment:
@speier Pretty sure you can drop the `window` here.

Reply:
✅